### ISSUE TITLE
ENOENT: no such file or directory, stat '.../tmp/funnel-output_path-yidKj7MR.tmp/package.json

### ISSUE THREAD

**Issue by jacobq** [ember-cli/ember-cli-update#451]
I tried to run `ember-cli-update` on an addon that was using ember-cli ~3.3.0 just now and was surprised to see an error message that appears related to the build process somehow. Any insight appreciated...

`$ DEBUG=ember-cli-update,git-diff-apply ember-cli-update`
<details><summary>Output</summary>

```
  ember-cli-update npm info ember-cli versions --json +0ms
  ember-cli-update [
  ember-cli-update   "0.0.0",
  ember-cli-update   "0.0.1",
  ember-cli-update   "0.0.2",
  ember-cli-update   "0.0.3",
  ember-cli-update   "0.0.4",
  ember-cli-update   "0.0.5",
  ember-cli-update   "0.0.6",
  ember-cli-update   "0.0.7",
  ember-cli-update   "0.0.8",
  ember-cli-update   "0.0.9",
  ember-cli-update   "0.0.10",
  ember-cli-update   "0.0.11",
  ember-cli-update   "0.0.12",
  ember-cli-update   "0.0.13",
  ember-cli-update   "0.0.14",
  ember-cli-update   "0.0.15",
  ember-cli-update   "0.0.16",
  ember-cli-update   "0.0.18",
  ember-cli-update   "0.0.19",
  ember-cli-update   "0.0.20",
  ember-cli-update   "0.0.21",
  ember-cli-update   "0.0.22",
  ember-cli-update   "0.0.23",
  ember-cli-update   "0.0.24",
  ember-cli-update   "0.0.25",
  ember-cli-update   "0.0.27",
  ember-cli-update   "0.0.28",
  ember-cli-update   "0.0.29",
  ember-cli-update   "0.0.32",
  ember-cli-update   "0.0.33",
  ember-cli-update   "0.0.34",
  ember-cli-update   "0.0.35",
  ember-cli-update   "0.0.36",
  ember-cli-update   "0.0.37",
  ember-cli-update   "0.0.38",
  ember-cli-update   "0.0.39",
  ember-cli-update   "0.0.40",
  ember-cli-update   "0.0.41",
  ember-cli-update   "0.0.42",
  ember-cli-update   "0.0.43",
  ember-cli-update   "0.0.44",
  ember-cli-update   "0.0.45",
  ember-cli-update   "0.0.46",
  ember-cli-update   "0.0.47",
  ember-cli-update   "0.1.0",
  ember-cli-update   "0.1.1",
  ember-cli-update   "0.1.2",
  ember-cli-update   "0.1.3",
  ember-cli-update   "0.1.4",
  ember-cli-update   "0.1.5",
  ember-cli-update   "0.1.6",
  ember-cli-update   "0.1.7",
  ember-cli-update   "0.1.8",
  ember-cli-update   "0.1.9",
  ember-cli-update   "0.1.10",
  ember-cli-update   "0.1.11",
  ember-cli-update   "0.1.12",
  ember-cli-update   "0.1.13",
  ember-cli-update   "0.1.14",
  ember-cli-update   "0.1.15",
  ember-cli-update   "0.2.0-beta.1",
  ember-cli-update   "0.2.0",
  ember-cli-update   "0.2.1",
  ember-cli-update   "0.2.2",
  ember-cli-update   "0.2.3",
  ember-cli-update   "0.2.4",
  ember-cli-update   "0.2.5",
  ember-cli-update   "0.2.6",
  ember-cli-update   "0.2.7",
  ember-cli-update   "1.13.0",
  ember-cli-update   "1.13.1",
  ember-cli-update   "1.13.5",
  ember-cli-update   "1.13.6",
  ember-cli-update   "1.13.7",
  ember-cli-update   "1.13.8",
  ember-cli-update   "1.13.9",
  ember-cli-update   "1.13.10",
  ember-cli-update   "1.13.11",
  ember-cli-update   "1.13.12",
  ember-cli-update   "1.13.13",
  ember-cli-update   "1.13.14",
  ember-cli-update   "1.13.15",
  ember-cli-update   "2.2.0-beta.1",
  ember-cli-update   "2.2.0-beta.2",
  ember-cli-update   "2.2.0-beta.3",
  ember-cli-update   "2.2.0-beta.4",
  ember-cli-update   "2.2.0-beta.5",
  ember-cli-update   "2.2.0-beta.6",
  ember-cli-update   "2.3.0-beta.1",
  ember-cli-update   "2.3.0-beta.2",
  ember-cli-update   "2.3.0",
  ember-cli-update   "2.4.0",
  ember-cli-update   "2.4.1",
  ember-cli-update   "2.4.2",
  ember-cli-update   "2.4.3",
  ember-cli-update   "2.5.0",
  ember-cli-update   "2.5.1",
  ember-cli-update   "2.6.0-beta.1",
  ember-cli-update   "2.6.0-beta.2",
  ember-cli-update   "2.6.0-beta.3",
  ember-cli-update   "2.6.0",
  ember-cli-update   "2.6.1",
  ember-cli-update   "2.6.2",
  ember-cli-update   "2.6.3",
  ember-cli-update   "2.7.0-beta.1",
  ember-cli-update   "2.7.0-beta.2",
  ember-cli-update   "2.7.0-beta.3",
  ember-cli-update   "2.7.0-beta.4",
  ember-cli-update   "2.7.0-beta.5",
  ember-cli-update   "2.7.0-beta.6",
  ember-cli-update   "2.7.0",
  ember-cli-update   "2.8.0-beta.1",
  ember-cli-update   "2.8.0-beta.2",
  ember-cli-update   "2.8.0-beta.3",
  ember-cli-update   "2.8.0",
  ember-cli-update   "2.9.0-beta.1",
  ember-cli-update   "2.9.0-beta.2",
  ember-cli-update   "2.9.0",
  ember-cli-update   "2.9.1",
  ember-cli-update   "2.10.0-beta.1",
  ember-cli-update   "2.10.0-beta.2",
  ember-cli-update   "2.10.0",
  ember-cli-update   "2.10.1",
  ember-cli-update   "2.11.0-beta.1",
  ember-cli-update   "2.11.0-beta.2",
  ember-cli-update   "2.11.0",
  ember-cli-update   "2.11.1",
  ember-cli-update   "2.12.0-beta.1",
  ember-cli-update   "2.12.0-beta.2",
  ember-cli-update   "2.12.0",
  ember-cli-update   "2.12.1",
  ember-cli-update   "2.12.2",
  ember-cli-update   "2.12.3",
  ember-cli-update   "2.13.0-beta.1",
  ember-cli-update   "2.13.0-beta.2",
  ember-cli-update   "2.13.0-beta.3",
  ember-cli-update   "2.13.0-beta.4",
  ember-cli-update   "2.13.0",
  ember-cli-update   "2.13.1",
  ember-cli-update   "2.13.2",
  ember-cli-update   "2.13.3",
  ember-cli-update   "2.14.0-beta.1",
  ember-cli-update   "2.14.0-beta.2",
  ember-cli-update   "2.14.0",
  ember-cli-update   "2.14.1",
  ember-cli-update   "2.14.2",
  ember-cli-update   "2.15.0-beta.1",
  ember-cli-update   "2.15.0-beta.2",
  ember-cli-update   "2.15.0",
  ember-cli-update   "2.15.1",
  ember-cli-update   "2.16.0-beta.1",
  ember-cli-update   "2.16.0-beta.2",
  ember-cli-update   "2.16.0",
  ember-cli-update   "2.16.1",
  ember-cli-update   "2.16.2",
  ember-cli-update   "2.17.0-beta.1",
  ember-cli-update   "2.17.0-beta.2",
  ember-cli-update   "2.17.0",
  ember-cli-update   "2.17.1",
  ember-cli-update   "2.17.2",
  ember-cli-update   "2.18.0-beta.1",
  ember-cli-update   "2.18.0-beta.2",
  ember-cli-update   "2.18.0",
  ember-cli-update   "2.18.1",
  ember-cli-update   "2.18.2",
  ember-cli-update   "3.0.0-beta.1",
  ember-cli-update   "3.0.0-beta.2",
  ember-cli-update   "3.0.0",
  ember-cli-update   "3.0.1",
  ember-cli-update   "3.0.2",
  ember-cli-update   "3.0.3",
  ember-cli-update   "3.0.4",
  ember-cli-update   "3.1.0-beta.1",
  ember-cli-update   "3.1.0",
  ember-cli-update   "3.1.1",
  ember-cli-update   "3.1.2",
  ember-cli-update   "3.1.3",
  ember-cli-update   "3.1.4",
  ember-cli-update   "3.2.0-beta.1",
  ember-cli-update   "3.2.0-beta.2",
  ember-cli-update   "3.2.0",
  ember-cli-update   "3.3.0",
  ember-cli-update   "3.4.0-beta.1",
  ember-cli-update   "3.4.0-beta.2",
  ember-cli-update   "3.4.0-beta.3",
  ember-cli-update   "3.4.1",
  ember-cli-update   "3.4.2-beta.1",
  ember-cli-update   "3.4.2",
  ember-cli-update   "3.4.3",
  ember-cli-update   "3.5.0-beta.1",
  ember-cli-update   "3.5.0-beta.2",
  ember-cli-update   "3.5.0",
  ember-cli-update   "3.5.1",
  ember-cli-update   "3.6.0-beta.1"
  ember-cli-update ]
  ember-cli-update  +402ms
  ember-cli-update npm info ember-cli@latest version --json +2ms
  ember-cli-update "3.5.1"
  ember-cli-update  +392ms
  git-diff-apply git status --porcelain +0ms
  git-diff-apply  +4ms
  git-diff-apply git clone --mirror https://github.com/ember-cli/ember-addon-output /tmp/tmp-19597AtkrK4oUa9jX/.git +0ms
Cloning into bare repository '/tmp/tmp-19597AtkrK4oUa9jX/.git'...
  git-diff-apply  +646ms
  git-diff-apply git --git-dir="/tmp/tmp-19597AtkrK4oUa9jX/.git" rev-parse v3.3.0 +0ms
  git-diff-apply e421b03501f86c15467c788a69908d5bc9df8d78
  git-diff-apply  +2ms
  git-diff-apply git --git-dir="/tmp/tmp-19597AtkrK4oUa9jX/.git" --work-tree="/tmp/tmp-19597AtkrK4oUa9jX" checkout e421b03501f86c15467c788a69908d5bc9df8d78 +0ms
Note: checking out 'e421b03501f86c15467c788a69908d5bc9df8d78'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at e421b03... 3.3.0
  git-diff-apply  +4ms
  git-diff-apply git --git-dir="/tmp/tmp-19597AtkrK4oUa9jX/.git" rev-parse v3.5.1 +2ms
  git-diff-apply 4f4fa16bfad77a52be82debf01a46b3c6f4389a1
  git-diff-apply  +2ms
  git-diff-apply git --git-dir="/tmp/tmp-19597AtkrK4oUa9jX/.git" --work-tree="/tmp/tmp-19597AtkrK4oUa9jX" checkout 4f4fa16bfad77a52be82debf01a46b3c6f4389a1 +0ms
Previous HEAD position was e421b03... 3.3.0
HEAD is now at 4f4fa16... 3.5.1
  git-diff-apply  +5ms
  git-diff-apply git rev-parse --show-toplevel +0ms
  git-diff-apply /home/user/project
  git-diff-apply  +2ms
  git-diff-apply git --git-dir="/tmp/tmp-19597AtkrK4oUa9jX/.git" rev-parse v3.3.0 +0ms
  git-diff-apply e421b03501f86c15467c788a69908d5bc9df8d78
  git-diff-apply  +3ms
  git-diff-apply git --git-dir="/tmp/tmp-19597AtkrK4oUa9jX/.git" --work-tree="/tmp/tmp-19597AtkrK4oUa9jX" checkout e421b03501f86c15467c788a69908d5bc9df8d78 +0ms
Previous HEAD position was 4f4fa16... 3.5.1
HEAD is now at e421b03... 3.3.0
  git-diff-apply  +4ms
  git-diff-apply chdir /home/user/project +0ms
  git-diff-apply git symbolic-ref --short HEAD +0ms
  git-diff-apply master
  git-diff-apply  +2ms
  git-diff-apply git checkout --orphan e83ddbd0-ed1e-11e8-aa88-6d7a35d4e56b +1ms
Switched to a new branch 'e83ddbd0-ed1e-11e8-aa88-6d7a35d4e56b'
  git-diff-apply  +3ms
  git-diff-apply git reset --hard +0ms
  git-diff-apply  +4ms
  git-diff-apply ensureDirSync /home/user/project +10ms
  git-diff-apply chdir /home/user/project +0ms
  git-diff-apply [ '.git', 'tmp' ] +0ms
  git-diff-apply moveAll /home/user/project /tmp/tmp-19597ZVk27JWKWvf6 +2ms
  git-diff-apply git checkout master +41ms
Switched to branch 'master'
  git-diff-apply Your branch is up-to-date with 'origin/master'.
  git-diff-apply  +8ms
{ [Error: ENOENT: no such file or directory, stat '/home/user/project/tmp/funnel-output_path-yidKj7MR.tmp/package.json']
  errno: -2,
  code: 'ENOENT',
  syscall: 'stat',
  path:
   '/home/user/project/tmp/funnel-output_path-yidKj7MR.tmp/package.json' }
```

</details>

<details><summary>Version info</summary>

```
$ ember -v
ember-cli: 3.5.1
node: 10.13.0
os: linux x64
```
</details>

**Comment by jacobq** [comment]
In fact, if you want to try running against the same repo you can: https://github.com/EmberMN/ember-cli-plotly

**Comment by kellyselden** [comment]
Looks like your project tmp dir is screwing it up somehow. Try deleting that and trying again.

**Comment by jacobq** [comment]
Thanks, Kelly. I can hardly believe that I hadn't already tried that, but sure enough that did the trick.

**Comment by kellyselden** [comment]
We can probably fix our codebase to not error in the first place.

**Comment by kellyselden** [comment]
This should be fixed in v0.32.5.
